src/ardrive.ts

```diff
@@ -39,23 +45,64 @@ export class ArDriveAnonymous extends ArDriveType {
 	}
 }
 
+// TODO: ArDrive should accept App-Name and App-Version tags from constructor?
+const commTipMetaTags: GQLTagInterface[] = [
+	{ name: 'App-Name', value: 'ArDrive-CLI' },
+	{ name: 'App-Version', value: '2.0' },
+	{ name: 'Trx-Type', value: 'Community-Tip' }
+];
+
 export class ArDrive extends ArDriveAnonymous {
-	constructor(protected readonly arFsDao: ArFSDAO) {
+	constructor(
+		private readonly wallet: Wallet,
+		private readonly walletDao: WalletDAO,
+		protected readonly arFsDao: ArFSDAO,
+		private readonly communityOracle: CommunityOracle
+	) {
 		super(arFsDao);
 	}
 
+	// TODO: FS shouldn't be reading the files more than once and doesn't belong in this class
+	getFileSize(filePath: string): Bytes {
+		return fs.statSync(filePath).size;
+	}
+
+	async sendCommunityTip(communityWinstonTip: Winston): Promise<ArFSTipData> {
+		const tokenHolder: ArweaveAddress = await this.communityOracle.selectTokenHolder();
+
+		const communityTipResult = await this.walletDao.sendARToAddress(
+			winstonToAr(+communityWinstonTip),
+			this.wallet,
+			tokenHolder,
+			commTipMetaTags
+		);
+
+		return { txId: communityTipResult.trxID, recipient: tokenHolder, winston: communityTipResult.winston };
```

Comment:
Let's update the ArFSTipData type to send back the entire transaction so that we can extract the network fee out for the ArDrive layer's return data (i.e. `trx.reward`).

Reply:
If we do that, we can probably get rid of all the other props here since it's all in the trx.

Reply:
I realize now that this comment's suggestion would have broken the architecture and leaked Arweave types into the upper layers.